vnext/Microsoft.ReactNative/Fabric/Composition/ReactNativeIsland.cpp

```diff
@@ -380,14 +380,38 @@ void ReactNativeIsland::InitRootView(
 
   m_context = winrt::Microsoft::ReactNative::ReactContext(std::move(context));
   m_reactViewOptions = std::move(viewOptions);
-  m_CompositionEventHandler = std::make_shared<::Microsoft::ReactNative::CompositionEventHandler>(m_context, *this);
+  m_CompositionEventHandler = std::make_shared<::Microsoft::ReactNative::CompositionEventHandler>(m_context, *this, -1);
   m_CompositionEventHandler->Initialize();
 
   UpdateRootViewInternal();
 
   m_isInitialized = true;
 }
 
+void ReactNativeIsland::AddFragmentCompositionEventHandler(
+    winrt::Microsoft::ReactNative::IReactContext context,
+    winrt::Microsoft::ReactNative::ComponentView componentView) noexcept {
+  m_uiDispatcher = context.Properties()
+                       .Get(winrt::Microsoft::ReactNative::ReactDispatcherHelper::UIDispatcherProperty())
+                       .try_as<IReactDispatcher>();
+  VerifyElseCrash(m_uiDispatcher.HasThreadAccess());
+  auto uiManager = ::Microsoft::ReactNative::FabricUIManager::FromProperties(
+      winrt::Microsoft::ReactNative::ReactPropertyBag(context.Properties()));
+
+  // currently Modal is only supported with one surface
+  m_rootTag = uiManager->getRootSurfaceTag();
```

Comment:
Is there a better way to get the m_rootTag of app? I used to be able to grab it through componentView's parent but the parent doesn't get set till after Modal is created now.

Reply:
You should get the roottag in onMount.

Having said that, I'm not sure about the usage of root tag here.  Putting the root tag here makes a bunch of the methods in ReactNativeIsland act on the root view, when it should be acting on the modal. 

Maybe the tag being set here should be componentView.Tag?

Reply:
I originally tried to use the componentView.Tag but since Modal isn't a RootComponentView, ReactNativeIsland would error out on GetComponentView because it traverses through a RootComponentView.

However I changed ModalComponent to be derived from RootComponentView and that seemed to fix things! Let me know if you think it will be an issue for a RNW instance to have multiple RootComponentViews though. I'm not seeing any errors currently on rntester but could be unaware of other situations.